posts/2025-07-03_swab-ww-comparison/index.qmd

```diff
@@ -0,0 +1,410 @@
+
+1. Simulate an exponentially growing outbreak.
+2. Specify a particular sampling scheme for swabs and wastewater:
+number of swabs in a pool, sequencing depth, read length, etc.
+3. For each sample from our posterior distribution of the previously fitted
+read count model, simulate the timecourse of viral reads spanning a junction.
+4. Use the simulations to calculate the joint distribution of cumulative incidence at detection in both sample types.
+
+This is meant as a quick first pass at the problem, not a definitive
+account of the merits of swabs vs. wastewater.
+
+# Setup
+
+```{r}
+#| label: setup
+#| message: false
+
+library(dplyr)
+library(tibble)
+library(readr)
+library(ggplot2)
+library(slider)
+library(knitr)
+library(kableExtra)
+
+set.seed(20250703)
+
+theme_set(theme_minimal())
+
+# Colors borrowed from https://github.com/JLSteenwyk/ggpubfigs
+wong_eight <- c(
+  "#E69F00",
+  "#56B4E9",
+  "#009E73",
+  "#F0E442",
+  "#0072B2",
+  "#D55E00",
+  "#CC79A7",
+  "#000000"
+)
+options(
+  ggplot2.discrete.colour = function()
+    scale_colour_manual(values = wong_eight)
+)
+options(
+  ggplot2.discrete.fill = function()
+    scale_fill_manual(values = wong_eight)
+)
+
+# For axis labels
+scientific <- function(x) {
+  parse(text = sprintf("10^{%f}", log10(x)))
+}
+
+# Geometric mean
+gm <- function(x) exp(mean(log(x)))
+```
+
+# Model
+
+## Epidemic
+
+We assume an epidemic that grows (deterministically) exponentially starting
+from one infected person in a population of 2.5M people (roughly the DITP sewershed).
+We stop the simulation when $\approx 50\%$ of people have been infected.
+We also assume that infected people shed the virus at a constant rate for a fixed
+number of days, starting with the day they are infected.
+
+```{r}
+#| fig-label: epi-model
+#| fig-cap: Epidemic progress by day
+
+# All times in days
+doubling_time <- 4
+growth_rate <- log(2) / doubling_time
+shedding_duration <- 7
+
+population_size <- 2.5e6
+# Grow until ~50% of people have been infected.
+# With growth rate r, cumulative incidence is approximately (1/r) exp(r t).
+# CI \approx N / 2 => t \approx log(N r / 2) / r
+max_day <- ceiling(log(growth_rate * population_size / 2) / growth_rate)
+
+epi_model <- tibble(
+  day = 0:max_day,
+  # Exponential growth from a single case
+  incidence = exp(day * growth_rate) / population_size,
+  cumulative_incidence = cumsum(incidence),
+  # Assume constant shedding for `shedding_duration` days
+  prevalence = slide_dbl(incidence, sum, .before = shedding_duration - 1, .complete = FALSE)
+)
+
+epi_model %>%
+  ggplot(aes(x = day)) +
+  geom_line(aes(y = cumulative_incidence, color = "Cumulative Incidence")) +
+  geom_line(aes(y = prevalence, color = "Prevalence")) +
+  labs(x = NULL, y = NULL, color = NULL)
+```
+
+
+## Sampling and sequencing
+
+* We assume that the wastewater and pooled nasal swabs are collected every day,
+and that each sample is sequenced to a constant number of total reads.
```

Comment:
minor suggestion to clarify that this constant number of total reads differs by sample type